oauth-client-credentials/README.md

```diff
@@ -0,0 +1,78 @@
+# Client credentials grant type
+
+This sample lets you request an OAuth token from Apigee using the OAuth 2.0 client credentials grant type flow. 
+
+## About client credentials
+
+Most typically, this grant type is used when the app is also the resource owner. For example, an app may need to access a backend cloud-based storage service to store and retrieve data that it uses to perform its work, rather than data specifically owned by the end user. This grant type flow occurs strictly between a client app and the authorization server. An end user does not participate in this grant type flow. 
+
+## How it works
+
+With the client credentials grant type flow, the client app requests an access token directly by providing its client ID and client secret keys. These keys are generated when you create a Developer App in Apigee. Apigee validates the credentials and returns an access token to the client. The client can then make secure calls to the resource server.
+
+The API is called like this, where the client ID and secret are Base64-encoded and used in the Basic Auth header:
+
+```
+curl -H "Authorization: Basic <base64-encoded key:secret>" https://your-api-url.com/oauth/token?grant_type=client_credentials
+```
+
+## Implementation on Apigee 
+
+The client credentials sample uses one policy that executes on Apigee : An OAuthV2 policy to generate the access token. The policy is attached to the `/token` endpoint (a custom flow on Apigee). 
+
+## Prerequisites
+
+To run this sample, you'll need:
+- [Apigee Organization](https://cloud.google.com/apigee/pricing)
+
+
+## Dependencies
+- [`apigee-sackmesser`](https://github.com/apigee/devrel/tree/main/tools/apigee-sackmesser)
```

Comment:
should we use apigeecli instead?

Reply:
Sure, that makes sense. I've added apigeecli and also reworked the sample to use the CloudShell.